frontend/online-shop/src/components/Products.tsx

```diff
@@ -0,0 +1,60 @@
+import { IconButton, ThemeProvider } from "@material-ui/core";
+import { useStyles } from "../styles/tableStyles";
+import InfoIcon from "@material-ui/icons/Info";
+import { Link } from "react-router-dom";
+import axios from "axios";
+import ProductService from "../services/ProductService";
+import React from "react";
+
+export interface ProductEntityProps {
+  id: number;
+  name: string;
+  description: string;
+  category: string;
+  price: number;
+  weight: number;
+  imageUrl: string;
+}
+
+export const TableProducts = (props: ProductEntityProps) => {
+  var productService = new ProductService();
+  var result = productService.getProducts();
+
+  const [data, setData] = React.useState<any[]>([]);
+
+  React.useEffect(() => {
+    result.then((res) => {
+      setData(res.data);
+    });
+  }, []);
+
+  const classes = useStyles();
+  const items = data.map((product) => (
+    <tr className={classes.trStyle}>
```

Comment:
React needs a key on the item elements within a map result. You could wrap everything with a `<Box key={product.id}>`

Reply:
Added the key value to the <tr> tag because using the box compressed everything in a cell only.